### PR TITLE
WIP: Extract ELF section reading from rt.backtrace.elf to core.elf

### PR DESCRIPTION
**Pull request by kinke**
Incl. some refactoring. Still needs to be verified; there are surely obsolete imports etc.

### CODE REVIEW

**Comment by kinke** [comment]
Pinging @joakim-noah.

File: src/rt/backtrace/elf.d [hunk 1]
```diff
@@ -45,21 +60,16 @@ struct Image
...
-            // debug_line section found and loaded
-            return ElfSection(&file, &dbgSectionHeader);
-        }
+        // we don't support compressed debug sections
+        if ((dbgSectionHeader.shdr.sh_flags & SHF_COMPRESSED) != 0)
+            return null;
 
-        return null;
+        // debug_line section found and loaded
+        return cast(const(ubyte)[]) ElfSection(file, dbgSectionHeader).get();
```
Comment (kinke) [inline src/rt/backtrace/elf.d]: This should be blatantly wrong, unmapping the section data right before returning the slice. I wonder why the original code seems to be working...

**Comment by joakim-noah** [comment]
Good idea, why isn't this submitted upstream?

**Comment by kinke** [comment]
Needs to be tested and polished first, and I'll probably do this here by simplifying the DSO registration further (no magic linker symbols for the `.minfo` section required anymore).

**Comment by joakim-noah** [comment]
OK, will this get in before the final 1.12 release?

**Comment by kinke** [comment]
Nope, definitely not.

**Comment by kinke** [comment]
The backtrace generation code is working fine on Linux x64.

I'll move some other pieces from `rt.sections_elf_shared` to the new module, in order to finally expose convenience functions like `const(void)[] getSectionByName(void* someAddressInDesiredDSO, const(char)* sectionName)`.

**Comment by kinke** [comment]
Now a branch in the official repo and tested by https://github.com/ldc-developers/ldc/pull/2875.

**Pull request closed without merging**